crates/abq_utils/src/net_protocol/runners.rs

```diff
@@ -289,38 +289,59 @@ impl From<RawManifest> for Manifest {
 impl Manifest {
     /// Flattens a manifest into [TestSpec]s, preserving the manifest order.
     pub fn flatten(self) -> (Vec<TestSpec>, MetadataMap) {
-        use v0_2::{Group, Test, TestCase, TestOrGroup};
-
+        use v0_2::{Group, TestOrGroup};
         let Manifest { members, init_meta } = self;
         let mut collected = Vec::with_capacity(members.len());
-        let mut queue: VecDeque<_> = members.into_iter().collect();
-        while let Some(test_or_group) = queue.pop_front() {
-            match test_or_group {
-                TestOrGroup::Test(Test { id, meta, .. }) => {
-                    let spec = TestSpec {
-                        // Generate a fresh ID for ABQ-internal usage.
-                        work_id: WorkId::new(),
-                        test_case: self::TestCase(
-                            TestCase {
-                                id,
-                                meta,
-                                focus: None,
-                            }
-                            .into(),
-                        ),
-                    };
-                    collected.push(spec);
+
+        // 🙋 Ayaz told me to reuse this Deque but I don't understand the implications of re-creating where we extend it below
+        let mut queue: VecDeque<v0_2::TestOrGroup> = VecDeque::new();
```

Comment:
You could even use a `Vec` directly, no need for a deque here actually (sorry)

Reply:
can I use `members` directly?